msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,165 @@
+| &nbsp; | [Microsoft Edge](https://www.microsoft.com/edge) (recommended) or [Google Chrome](https://www.google.com/chrome/) | A browser with developer tools. |
+| &nbsp; | [Visual Studio Code](https://code.visualstudio.com/download) | JavaScript, TypeScript, or SharePoint Framework (SPFx) build environments. Use the latest version. |
+
+To build Actions for your app, follow these steps:
+
+> [!div class="checklist"]
+>
+>* [Update app manifest](#update-app-manifest).
+>* [Retrieve Action information through context object](#retrieve-action-information-through-context-object).
+>* [Access content through Graph API](#access-content-through-graph-api).
+
+## Update app manifest
+
+Define the intent, object, and handler for your actions in the app manifest (previously called Teams app manifest).
+
+The following is an example of the `intent`, `supportedobjects`, and `handlers` properties for `openPage` in app manifest:
+
+```json
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/teams/vDevPreview/MicrosoftTeams.schema.json",
+  "manifestVersion": "devPreview",
+.
+.
+.
+"actions": [
+    {
+        // Defining Action to open a page
+        "id": "relatedTasks",
+        "displayName": "Related tasks",
+        "intent": "custom",
+        "description": "Shows tasks in the To do app that are related to this file.",
+        "handlers": [
+            {
+                "type": "openPage",
+                "supportedObjects": {
+                    "file": {
+                        "extensions": ["xlsx", "doc", "docx", "pdf", "pptx", "ppt"]
+                    }
+                },
+                "pageInfo": {
+                    "pageId": "index",
+                }
+            }
+        ]
+    }
+]
+}
+```
+
+For more information, see [public developer preview app manifest schema](../resources/schema/manifest-schema-dev-preview.md#actions).
+
+When a user selects an action to open a page and view related tasks in an app based on the selected file. The app uses the `"intent": "custom"` property to identify the file type, such as .xlsx or doc and the `"type": "openPage"` handler opens the app and navigates to the `pageId`.
+
+## Retrieve Action information through context object
+
+Update the handler to receive the Action information through the context object to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
+
+When a user selects Add option from the app's context menu, a page opens with the help of the `openPage` property in the app manifest. Your app can access contextual information about the invoked Action from the `actionInfo` property of the context object `app.getContext()`.
+
+```javascript
```

Comment:
Remove this code snippet, since it should run after the lower snippet. I suggest to merge the main line into the lower section. See lower code snippet section